core/controllers/review_tests.py

```diff
@@ -0,0 +1,98 @@
+
+from constants import constants
+from core.controllers import acl_decorators
+from core.controllers import base
+from core.domain import dependency_registry
+from core.domain import interaction_registry
+from core.domain import obj_services
+from core.domain import story_services
+import feconf
+import jinja2
+
+
+class ReviewTestsPage(base.BaseHandler):
+    """Renders the review tests page."""
+
+    @acl_decorators.can_access_story_viewer_page
+    def get(self, story_id):
+        """Handles GET requests."""
+
+        if not constants.ENABLE_NEW_STRUCTURE_PLAYERS:
+            raise self.PageNotFoundException
+
+        story = story_services.get_story_by_id(story_id)
+
+        if story is None:
+            raise self.PageNotFoundException(
+                Exception('The story with the given id doesn\'t exist.'))
+
+        interaction_ids = feconf.ALLOWED_QUESTION_INTERACTION_IDS
+
+        interaction_dependency_ids = (
+            interaction_registry.Registry.get_deduplicated_dependency_ids(
+                interaction_ids))
+        dependencies_html, additional_angular_modules = (
+            dependency_registry.Registry.get_deps_html_and_angular_modules(
+                interaction_dependency_ids))
+
+        interaction_templates = (
+            interaction_registry.Registry.get_interaction_html(
+                interaction_ids))
+
+        self.values.update({
+            'DEFAULT_OBJECT_VALUES': obj_services.get_default_object_values(),
+            'additional_angular_modules': additional_angular_modules,
+            'INTERACTION_SPECS': interaction_registry.Registry.get_all_specs(),
+            'interaction_templates': jinja2.utils.Markup(
+                interaction_templates),
+            'dependencies_html': jinja2.utils.Markup(dependencies_html),
+            'story_name': story.title
+        })
+        self.render_template('dist/review-test-page.mainpage.html')
+
+
+class ReviewTestsPageDataHandler(base.BaseHandler):
+    """Fetches relevant data for the review tests page."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.can_access_story_viewer_page
+    def get(self, story_id):
+        """Handles GET request."""
+        if not constants.ENABLE_NEW_STRUCTURE_PLAYERS:
+            raise self.PageNotFoundException
+
+        story = story_services.get_story_by_id(story_id)
+        latest_completed_node_ids = (
+            story_services.get_latest_completed_node_ids(self.user_id, story_id)
+        )
+
+        if story is None:
+            raise self.PageNotFoundException(
+                Exception('The story with the given id doesn\'t exist.'))
+        if len(latest_completed_node_ids) == 0:
+            raise self.PageNotFoundException
+
+        acquired_skill_ids = story.get_acquired_skill_ids_for_node_ids(
+            latest_completed_node_ids
+        )
+
+        self.values.update({
+            'skill_list': acquired_skill_ids
```

Comment:
skill_list --> skill_ids, or skill_ids_to_test, or something similar. Otherwise, sounds like a list of Skill objects.

Reply:
Done.